### PR TITLE
libx265 3.0

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **libx265/3.0**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libx265/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                    "list(APPEND PLATFORM_LIBS pthread)", "")
+                tools.replace_in_file(os.path.join('sources', 'source', 'CMakeLists.txt'),
+                    "list(APPEND PLATFORM_LIBS rt)", "")
+            cmake = CMake(self, set_cmake_flags=True)
+            cmake.definitions['ENABLE_SHARED'] = self.options.shared
+            cmake.definitions['ENABLE_LIBNUMA'] = False
+            if self.settings.os == "Macos":
+                cmake.definitions['CMAKE_SHARED_LINKER_FLAGS'] = '-Wl,-read_only_relocs,suppress'
+            if self.settings.os != 'Windows':
+                cmake.definitions['CMAKE_POSITION_INDEPENDENT_CODE'] = self.options.fPIC
```
Comment (danimtb) [inline recipes/libx265/all/conanfile.py]: this is not needed

File: recipes/libx265/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                cmake.definitions['ENABLE_PIC'] = self.options.fPIC
+            cmake.definitions['HIGH_BIT_DEPTH'] = self.options.bit_depth != 8
+            cmake.definitions['MAIN12'] = self.options.bit_depth == 12
+            cmake.definitions['ENABLE_HDR10_PLUS'] = self.options.HDR10
+            if self.settings.os == "Linux":
+                cmake.definitions["PLATFORM_LIBS"] = "dl"
+            cmake.configure()
+            cmake.build()
+            cmake.install()
```
Comment (danimtb) [inline recipes/libx265/all/conanfile.py]: can we have this in the package method?

**Comment by ericLemanissier** [comment]
That's https://github.com/conan-io/conan-center-index/issues/624
There is nothing I can do about it

**Comment by ericLemanissier** [comment]
superseded by https://github.com/conan-io/conan-center-index/pull/819

**Comment by danimtb** [comment]
Thanks, @ericLemanissier. Seems the other recipe didn't need the ninja package to build so it avoided the issue reported here. Thanks anyway for the contribution, hope that recipe suits your needs as well

**Comment by ericLemanissier** [comment]
No problem, don't worry. I'm not sure how @madebr got rid of nasm and ninja. Or maybe they are not mandatory ?

**Comment by madebr** [comment]
@ericLemanissier Sorry I didn't see this pr when opening mine.
I just ported the bincrafters package and forgot to copy these build requirements.
The resulting package probably won't contain assembly optimizations so it would still be useful to add nasm as a build requirement (eventually hidden behind an option).

**Pull request closed without merging**